tap-agent/src/tap/sender_accounts_manager.rs

```diff
@@ -29,19 +29,19 @@ pub struct NewReceiptNotification {
     pub value: u128,
 }
 
-pub struct SenderAllocationRelationshipsManager {
+pub struct SenderAccountsManager {
     _inner: Arc<Inner>,
     new_receipts_watcher_handle: tokio::task::JoinHandle<()>,
     _eligible_allocations_senders_pipe: PipeHandle,
 }
 
-#[derive(Clone)]
+/// Inner struct for SenderAccountsManager. This is used to store an Arc state for spawning async
+/// tasks.
 struct Inner {
     config: &'static config::Cli,
     pgpool: PgPool,
-    /// Map of (allocation_id, sender_address) to SenderAllocationRelationship.
-    sender_allocation_relationships:
-        Arc<RwLock<HashMap<(Address, Address), SenderAllocationRelationship>>>,
+    /// Map of sender_address to SenderAllocation.
+    sender_accounts: Arc<RwLock<HashMap<Address, SenderAccount>>>,
```

Comment:
Could you make sure that we need RwLock from tokio?

Reply:
Yup, seems a bit overkill indeed. It is however read very often (on each receipt notification), and written every time the status from the graph's subgraph is updated.
However, I'm thinking that it doesn't really matter to actually lock, just to make sure that the `new_receipts_watcher` gets a fairly recent view of `sender_accounts`.

So here's my creative attempt: 69b06ddc5ae3e5d230fa90350b51c1185a1307e1
I'm copying the `sender_accounts` hashmap and replacing it, so that on the next loop of `new_receipts_watcher`, it will see a ref to the new updated hashmap.